apache/metron-bro-plugin-kafka
# METRON-2020: Running run_end_to_end.sh with docker give warning if bash < 4.0

## Contributor Comments
Added a check to the `analyze_results.sh` and `run_end_to_end.sh` scripts to verify the scripts are being run with bash >= 4.  Technically this is only needed due to the associative arrays in `analyze_results.sh` but I pushed it to `run_end_to_end.sh` as well to provide faster feedback since we know the analyzation at the end will fail if they're running bash 3 or earlier.

### Testing
- Run with bash < 4 (`/bin/bash` on a mac) and expect failure.
- Run with bash > 4 and expect success.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron's Bro kafka writer plugin.

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:

### For code changes:
- [X] Have you ensured that the full suite of tests and checks have been executed via:
  ```
  bro-pkg test $GITHUB_USERNAME/metron-bro-plugin-kafka --version $BRANCH
  ```
- [ ] Have you verified the basic functionality of the build by building and running locally with Apache Metron's [Vagrant full-dev environment](https://github.com/apache/metron/tree/master/metron-deployment/development/centos6) or the equivalent?
